crates/bindings/README.md

```diff
@@ -0,0 +1,517 @@
+
+```rust
+use spacetimedb::{reducer, ReducerContext};
+
+#[derive(Debug)]
+enum GiveItemError {
+    NoSuchItem(u64),
+    NonTransferable(u64)
+}
+
+#[reducer]
+fn give_player_item(
+    ctx: &ReducerContext,
+    player_id: u64,
+    item_id: u64
+) -> Result<(), GiveItemError> {
+    /* ... */
+}
+```
+
+Every reducer runs inside a [database transaction](https://en.wikipedia.org/wiki/Database_transaction). <!-- TODO: specific transaction level guarantees. --> This means that reducers will not observe the effects of other reducers modifying the database while they run. Also, if a reducer fails, all of its changes to the database will automatically be rolled back. Reducers can fail by [panicking](::std::panic!) or by returning an `Err`.
+
+#### The `ReducerContext` Type
+
+Reducers have access to a special [`ReducerContext`] argument. This argument allows reading and writing the database attached to a module. It also provides some additional functionality, like generating random numbers and scheduling future operations.
+
+The most important field of [`ReducerContext`] is [`.db`](ReducerContext#structfield.db). This field provides a view of the module's database. The [`#[table]`](macro@crate::table) macro generates traits that add accessor methods to this field.
+
+<!-- TODO: this seems to work sometimes, but not always... Sometimes the links to downstream trait implementations aren't generated for some reason. Maybe it only works in the same cargo workspace?
+
+To see all of the available methods on `ctx.db`, run `cargo doc` in your module's directory, and navigate to the `spacetimedb::Local` struct in the generated documentation. This will be at the path:
+- `[your_project_directory]/target/doc/spacetimedb/struct.Local.html` (non-Windows)
+- `[your_project_directory]\target\doc\spacetimedb\struct.Local.html` (Windows)
+-->
+
+#### The `log` crate
+
+SpacetimeDB Rust modules have built-in support for the [log crate](log). All modules automatically install a suitable logger when they are first loaded by SpacetimeDB. (At time of writing, this happens [here](https://github.com/clockworklabs/SpacetimeDB/blob/e9e287b8aab638ba6e8bf9c5d41d632db041029c/crates/bindings/src/logger.rs)). Log macros can be used anywhere in module code, and log outputs of a running module can be inspected using the `spacetime logs` command:
+
+```text
+spacetime logs <DATABASE_IDENTITY>
+```
+
+<!-- TODO: we seem to have removed `println`, is that still hidden somewhere? -->
+
+#### Lifecycle Reducers
+
+A small group of reducers are called at set points in the module lifecycle. These are used to initialize
+the database and respond to client connections. See [Lifecycle Reducers](macro@crate::reducer#lifecycle-reducers).
+
+#### Scheduled Reducers
+
+Reducers can be scheduled to run repeatedly. This can be used to implement timers, game loops, and
+maintenance tasks. See [Scheduled Reducers](macro@crate::reducer#scheduled-reducers).
```

Comment:
Is scheduled-ness a feature of the reducer, or of the table? If the latter, we may want to document this as "scheduled tables" or something. I'm unsure on what's best.

Reply:
I'm deciding I don't really care. This is fine.